### PR TITLE
Servers side callbacks

### PR DESCRIPTION
**Pull request by krystian8207**
This is version of https://github.com/Appsilon/shiny.router/pull/36, customized with our current dev changes.

You can see how it's working in `basic` and `server_overwrite` examples.

### CODE REVIEW

File: R/router.R [hunk 1]
```diff
@@ -1,18 +1,37 @@
 ROUTER_UI_ID <- '_router_ui'
 
+#' Create a mapping bewtween a ui element
+#' and a server callack
+#'
+#' @param ui Valid Shiny user interface.
+#' @param server Function that is called within the global server function if given
+callback_mapping <- function(ui, server = NA) {
+  server <- if (is.function(server)) {
```
Comment (dokato) [inline R/router.R]: what about `ifelse` here?

File: examples/server_overwrite/app.R [hunk 2]
```diff
@@ -0,0 +1,66 @@
+library(shiny)
+library(shiny.router)
+
+# This generates menu in user interface with links.
+menu <- (
+  tags$ul(
+    tags$li(a(class = "item", href = route_link("/"), "Page")),
+    tags$li(a(class = "item", href = route_link("second"), "Second page")),
+    tags$li(a(class = "item", href = route_link("third"), "A third page")),
+    tags$li(a(class = "item", href = route_link("fourth"), "A fourth page"))
```
Comment (dokato) [inline examples/server_overwrite/app.R]: What is the purpose of page no 4 in this example?

Comment (krystian8207) [reply examples/server_overwrite/app.R]: This was case when there is no specified local server callback (but global does) and the page is not loaded on app start. Very similiar to the main page but worth to realize the behaviour.

File: R/router.R (hunk 1 above)
Comment (krystian8207) [reply R/router.R]: I think ifelse should be used to perform vectorisation. `callback_mapping` is not dedicated to it I suppose, see 16 and 17 line. Is there any other reason to do it?

Comment (dokato) [reply R/router.R]: Yes, agree. I prefer to use it sometimes for neater styling though. In that case NVM.

**Comment by dokato** [comment]
Ready to MRG

File: R/router.R (hunk 1 above)
Comment (neumanrq) [reply R/router.R]: 👍 I have no strong opinion on that, both variants are arguable to me. Let's roll with the one you feel most comfortable with!

**Pull request merged**